### DISCUSSION TITLE
Configuring WalletConnect Ethereum Provider optional chains

### DISCUSSION THREAD

**Discussion by r3lays** [wagmi-dev/references#468] [Q&A]
In the [Wagmi documentation](https://wagmi.sh/core/connectors/walletConnect), it's mentioned that they wrap the WalletConnect Ethereum Provider. However, there's a lack of details on how to configure the options for this provider, particularly regarding the optionalChains and optionalNamespaces configurations.

 Are we understanding this correctly? For our use case we must specify all of our chains as optional for WalletConnect because some wallets only work for specific chains and if a chainId is passed and the session is initialized, the wallet wont properly connect.

**Comment by glitch-txs** [comment]
Currently the WalletConnect connector requires at least one required chain, you can try to workaround it by passing the chainId to the connect function. currently some wallets won't support empty required namespaces, we'll update the connector once they do. Alternatively you can copy paste the connector on a ts file and modify it for the moment

**Reply by ElvisKrop** [reply]
Actually, you are a bit wrong. This is a comment from wallet connect repo about chains and optional chains - https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/providers/ethereum-provider/src/EthereumProvider.ts#L190-L194
One of these arguments can be undefined. I have created an issue(https://github.com/wagmi-dev/references/issues/469) and opened a PR (https://github.com/wagmi-dev/references/pull/470) to fix this behaviour.

**Reply by ElvisKrop** [reply]
oh, you are talking about wagmi. yes, it requires at least one chain for initialization
Hopefully, my PR will be soon reviewed to improve this connector

**Reply by glitch-txs** [reply]
the PR looks good, also check that to the connect function is passed some chain params too so there might be need of some changes in that part as well. We're currently waiting for some wallets to support empty required namespaces (some of them currently break with this config).
